photon-ml/src/main/scala/com/linkedin/photon/ml/avro/AvroUtils.scala

```diff
@@ -191,32 +192,39 @@ object AvroUtils {
    * Convert the Avro record of type [[BayesianLinearModelAvro]] to the model type [[GeneralizedLinearModel]]
    *
    * @param bayesianLinearModelAvro The input Avro record
-   * @param nameAndTermToIntMap The map from feature name of type [[NameAndTerm]] to feature index of type [[Int]]
+   * @param featureMap The map from feature name of type [[NameAndTerm]] to feature index of type [[Int]]
    * @return The generalized linear model converted from the Avro record
    */
   protected[avro] def convertBayesianLinearModelAvroToGLM(
       bayesianLinearModelAvro: BayesianLinearModelAvro,
-      featureMap: IndexMap): GeneralizedLinearModel = {
+      featureMap: Option[IndexMap]): GeneralizedLinearModel = {
 
     val meansAvros = bayesianLinearModelAvro.getMeans
     val modelClass = bayesianLinearModelAvro.getModelClass.toString
     val indexAndValueArrayBuffer = new mutable.ArrayBuffer[(Int, Double)]
 
+    var counter: Int = 0
+
     val iterator = meansAvros.iterator()
     while (iterator.hasNext) {
       val feature = iterator.next()
       val name = feature.getName.toString
       val term = feature.getTerm.toString
       val featureKey = Utils.getFeatureKey(name, term)
-      if (featureMap.contains(featureKey)) {
+      if (featureMap.isDefined && featureMap.get.contains(featureKey)) {
         val value = feature.getValue
-        val index = featureMap.getOrElse(featureKey,
+        val index = featureMap.get.getOrElse(featureKey,
           throw new NoSuchElementException(s"nameAndTerm $featureKey not found in the feature map"))
```

Comment:
Why do we have this check if we're doing an explicit `contains` check beforehand? As I understand, currently this will never be called and instead we will ignore features not found in the index.